### ISSUE TITLE
Imports from @react-hooks/test failing with circular dependency TypeError

### ISSUE THREAD

**Issue by kevinhaube** [coinbase/rest-hooks#1689]
**React version** (e.g., 16.8.5)
17.0.2

**Concurrent mode** yes/no
no

**Rest Hooks version** (e.g., 4.5.9)
```json
"@rest-hooks/core": "^3.0.2",
"@rest-hooks/endpoint": "^2.1.0",
"@rest-hooks/hooks": "^2.1.0",
"@rest-hooks/rest": "^3.0.1",
"@rest-hooks/test": "^7.1.3",
```

**Describe the bug**
Any and all imports from `@rest-hooks/test` cause the following error: `TypeError: Class extends value undefined is not a constructor or null`. 

I first got this error when using `MockResolver` in my component unit tests. The line that failed was not my usage, but rather my import of it. A fair amount of searching led me to believe this is related to circular dependencies or a failure for a dependency to load. After ensuring the integrity of my own `.test.tsx` and `.tsx` files, I figured I'd try another approach: I tried the deprecated `MockProvider`, as well as using `mockInitialState()`, all of which resulted in the same import error, not usage errors.

**To Reproduce**
1. Follow the instructions here to set up `MockResolver` for a component unit test. (I'm using Jest)
2. Run tests 

**Expected behavior**
The packages should import without error.

**Additional context**
I tried these imports in a separate file, as well, and received the same failure. I followed `react-testing-library`'s approach to creating custom renderers, and created one with a `MockResolver` wrapping the `children`, and the import once again failed.

**Comment by ntucker** [comment]
Can you share your unit testing setup - like node version, jest version, etc

**Comment by ntucker** [comment]
Alternatively if you have a minimum reproducible example that you should share it would help greatly in identifying the issue.

**Comment by ntucker** [comment]
Do you have a stack trace for `TypeError: Class extends value undefined is not a constructor or null`? I cannot reproduce this

**Comment by ntucker** [comment]
I stumbled across a export type in https://github.com/coinbase/rest-hooks/pull/1701. Can you try with 7.2.0 @rest-hooks/test release and see if that helps? @kevinhaube

**Comment by ntucker** [comment]
I'm gonna assume this is fixed; feel free to open another ticket if you have any more issues. Thanks!

**Comment by ntucker** [comment]
Figured out the issue by checking out https://github.com/CDCgov/prime-reportstream/pull/4235

For the longest time I couldn't figure out why there were only two exports in rest hooks and furthermore it wasn't even running the JS in there. Then I found this:

```tsx
jest.mock("rest-hooks", () => ({
    useResource: () => {
        return mockData;
    },
    /* Must return children when mocking, otherwise nothing inside renders *
    NetworkErrorBoundary: ({ children }: { children: JSX.Element[] }) => {
        return <>{children}</>;
    },
}));
```

So yeah, it's trying to extend from a class that was explicitly deleted.

@rest-hooks/test exists so you don't have to do dangerous and hard to track things like mock out modules. There's a high likelihood of weird nasty stuff like this to show up.
